### DISCUSSION TITLE
having trouble generating a structure for discord webhooks

### DISCUSSION THREAD

**Discussion by NekoRem** [nlohmann/json#3783] [Q&A]
i am trying to use json for a discord webhook and i am trying to create an embed and I'm certain I'm creating the json structure incorrectly..
![Screenshot 2022-10-03 235852](https://user-images.githubusercontent.com/31341992/193701497-610c77ee-a75c-47dc-adff-a55ff8b2bae9.png) this image is what is wrong (bottom line is the returned value from discord api)
![Screenshot 2022-10-03 235633](https://user-images.githubusercontent.com/31341992/193701498-83e7c3c7-e6b1-45ff-9bab-8dc5424b6227.png) this image is correct
![Screenshot 2022-10-04 000842](https://user-images.githubusercontent.com/31341992/193701660-1a8cf129-0d23-4563-9196-574bcb92d468.png) this is the code I'm using to generate my structure

**Comment by nlohmann** [comment]
Not enough information to diagnose this — the post references two images (the wrong vs. correct payload) and 'the code I'm using', but no actual text/JSON/code was included in the discussion body, so there's nothing concrete to check. In general, a broken Discord embed webhook is most often caused by putting embed fields directly on the top-level webhook JSON instead of nesting them inside an `"embeds": [ { ... } ]` array, or by making `"fields"` something other than an array of `{"name": ..., "value": ..., "inline": ...}` objects. Re-posting the actual JSON string your code produces (via j.dump()) alongside the JSON Discord expects would let this be diagnosed concretely.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*

**Reply by NekoRem** [reply]
Bro replied after 4 years haha

**Reply by nlohmann** [reply]
Better late than never...
